libqtile/config.py

```diff
@@ -569,9 +569,12 @@ class Match:
     net_wm_pid:
         matches against the _NET_WM_PID atom (only int allowed for this
         rule)
+    func:
+        matches against the given function, which receives the client as only
+        argument
```

Comment:
Patch itself seems fine, but can we give this an explicit doc about what args the function takes and what it should return? And maybe a `Callable[[window.Window], bool]` annotation.

Reply:
> Patch itself seems fine, but can we give this an explicit doc about what args the function takes and what it should return?

Here or somewhere else?

Reply:
> And maybe a `Callable[[window.Window], bool]` annotation.

Added

Reply:
> Here or somewhere else?

Here, the current docstring for the function is mostly useless, IMO :). Before the type annotation, I had to read the source to figure out what it got as an argument and what it should return.

Reply:
done

Reply:
done